src/utils/getSearchParamsFromRequest.js

```diff
@@ -1,26 +1,23 @@
-export function getSearchParamsFromRequest(request, defaultValues = {}, transform = null) {
+export function getSearchParamsFromRequest(request, defaultValues = {}, transform = it => it) {
   if (!request?.url) return {};
   try {
-    const url = new URL(request.url);
     const searchParams = {};
+    const url = new URL(request.url);
+
     url.searchParams.forEach((val, key) => {
-      if (searchParams[key]) {
-        if (Array.isArray(searchParams[key])) {
-          searchParams[key].push(val);
-        } else {
-          const currValue = searchParams[key];
-          searchParams[key] = [currValue, val];
-        }
-      } else {
+      if (!searchParams[key]) {
         searchParams[key] = val;
+      } else {
+        if (!Array.isArray(searchParams[key])) {
+          searchParams[key] = [searchParams[key]];
+        }
+        searchParams[key].push(val);
```

Comment:
@Yivgha can you check this branch one more time? for me it looks that you are writing twice to the array in case of `searchParams[key]` is not an array.

Reply:
i've updated it a bit to be more clear, but seems like its okay 🤔